kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ListAllReassignmentsActionIntegrationTest.java

```diff
@@ -0,0 +1,90 @@
+import io.confluent.kafkarest.integration.ClusterTestHarness;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.apache.kafka.clients.admin.NewPartitionReassignment;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ListAllReassignmentsActionIntegrationTest extends ClusterTestHarness {
+
+  private static final String TOPIC_NAME = "topic-1";
+
+  public ListAllReassignmentsActionIntegrationTest() {
+    super(/* numBrokers= */ 6, /* withSchemaRegistry= */ false);
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    Map<Integer, List<Integer>> replicaAssignments = createAssignment(Arrays.asList(0, 1, 2));
+    createTopic(TOPIC_NAME, replicaAssignments);
+  }
+
+  @Test
+  public void listAllReassignments_returnsReassignments() throws Exception {
+    String clusterId = getClusterId();
+
+    Map<TopicPartition, Optional<NewPartitionReassignment>> reassignmentMap =
+        createReassignment(Arrays.asList(3, 4, 5));
+
+    alterPartitionReassignment(reassignmentMap);
+
+    Response response = request("/v3/clusters/" + clusterId + "/topics/-/partitions"
+        + "/-/reassignments")
+        .accept(Versions.JSON_API)
+        .get();
+    assertEquals(Status.OK.getStatusCode(), response.getStatus());
+
+    List<ReassignmentData> actualReassignments =
+        response.readEntity(ListReassignmentsResponse.class).getData();
+    for (ReassignmentData data : actualReassignments) {
+      assertEquals(data.getAttributes().getAddingReplicas(),
+          reassignmentMap.get(new TopicPartition(TOPIC_NAME,
+              data.getAttributes().getPartitionId())).get().targetReplicas());
+    }
+  }
+
+  @Test
+  public void listAllReassignments_nonExistingCluster_returnsNotFound() throws Exception {
+
+    Response response = request("/v3/clusters/foobar/topics/-/partitions"
+        + "/-/reassignments")
+        .accept(Versions.JSON_API)
+        .get();
+
+    assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+  }
+
+
+  private Map<Integer, List<Integer>> createAssignment(List<Integer> replicaIds) {
+    Map<Integer, List<Integer>> replicaAssignments = new HashMap<>();
+    for (int i = 0; i < 100; i++) {
+      replicaAssignments.put(i, replicaIds);
+    }
+    return replicaAssignments;
+  }
+
+  private Map<TopicPartition, Optional<NewPartitionReassignment>> createReassignment(
+      List<Integer> replicaIds) {
+    Map<TopicPartition, Optional<NewPartitionReassignment>> reassignmentMap = new HashMap<>();
+    for (int i = 0; i < 100; i++) {
+      reassignmentMap.put(new TopicPartition(TOPIC_NAME, i),
+          Optional.of(new NewPartitionReassignment(replicaIds)));
+    }
+    return reassignmentMap;
+  }
```

Comment:
As these methods will be used in several places, will create a util class in a follow up PR.

Reply:
You can add them to `ClusterHarness`.